cla_backend/apps/cla_butler/management/commands/monitor_missing_outcome_codes.py

```diff
@@ -0,0 +1,50 @@
+# coding=utf-8
+import logging
+from django.core.management.base import BaseCommand
+
+from cla_butler.stack import is_first_instance, InstanceNotInAsgException, StackException
+
+from cla_eventlog.constants import LOG_LEVELS, LOG_TYPES
+from cla_eventlog.models import Log
+from legalaid.models import Case
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = 'Count and alert when outcome codes denormalized to Case instances are missing'
+
+    def handle(self, *args, **options):
+        if self.should_run_housekeeping(**options):
+            self.check_for_missing_outcome_codes()
+        else:
+            self.stdout.write('Not doing housekeeping because running on secondary instance')
+
+    @staticmethod
+    def check_for_missing_outcome_codes():
+        outcomes_that_should_be_denormed = Log.objects.filter(level=LOG_LEVELS.HIGH, type=LOG_TYPES.OUTCOME)
+        outcomes_missing_denormed_code = outcomes_that_should_be_denormed.filter(case__outcome_code='')
+        cases_missing_denormed_code_pks = outcomes_missing_denormed_code.values_list('case__id', flat=True)
```

Comment:
Depending on how/when this command runs, would the above query be efficient or do we need to time it to less busy periods to avoid disruption to the operator's work?

Reply:
It's a small query on smallish amount of data. But reworked for efficiency. Worth logging resource usage and initially running only out of hours at first?

Reply:
Seeing that the linked cronjob runs at 1 am, I think that is fine. 👍